### PR TITLE
Hide toc headers

### PR DESCRIPTION
**Pull request by JKarlavige**
[Asana task](https://app.asana.com/0/1200099998847559/1205976870784519/f)

## What are you changing in this pull request and why?

If h2 or h3 headers exist within the Tabs or Expandable components, they appear in the TOC. However they are unable to link to the section in the page from the TOC as they are hidden elements initially.

This filters out TOC headers which are nested in Tabs or Expandable components. 

## Previews

This page has tabs and a demo `expandable` component at the top of the page. Verify the headers within these sections are removed from the TOC:
https://docs-getdbt-com-git-hide-toc-headers-dbt-labs.vercel.app/docs/cloud/cloud-cli-installation

This page has h2 & h3 headings, verify they remain on the page:
https://docs-getdbt-com-git-hide-toc-headers-dbt-labs.vercel.app/docs/cloud/about-cloud/architecture

## Checklist
- [x] Remove demo Expandable content from cloud-cli-installation page

### CODE REVIEW

File: website/docs/docs/cloud/cloud-cli-installation.md
Comment (JKarlavige) [inline website/docs/docs/cloud/cloud-cli-installation.md:1]: Note - remove all edits on this page before merge

**Review by mirnawong1 — APPROVED** [review]
this looks good to me @JKarlavige ! however one small note is that the toc displays the headers temporarily when the page loads and then it disappears again.  if you go to the URL (e.g. https://docs-getdbt-com-git-hide-toc-headers-dbt-labs.vercel.app/docs/cloud/cloud-cli-installation?install=pip#test-expandable-with-header), it doesn't take you to the section. but i think that's general behavior. other than that, looking great! thank you so much!

Screenshot 2024-03-13 at 15 56 49
Screenshot 2024-03-13 at 15 57 06

**Comment by mirnawong1** [comment]
works in mobile too!

**Comment by JKarlavige** [comment]
> however one small note is that the toc displays the headers temporarily when the page loads and then it disappears again.

yeah this was something i pointed out to our team as well with this. we can either leave it as-is, or hide the TOC (or show a loader) until everything is updated. i'm hesitant to do so as that would be a significant change which will need a good amount of testing.

fortunately i don't think this scenario is super common, but could definitely be improved.

> if you go to the URL (e.g. https://docs-getdbt-com-git-hide-toc-headers-dbt-labs.vercel.app/docs/cloud/cloud-cli-installation?install=pip#test-expandable-with-header), it doesn't take you to the section.

yes this is correct. it should be possible to update the expandable component to check the url for any params, and if a param in the url matches a header within the expandable component we can open the expandable component by default. but that could be spun up into its own task if you'd like us to update that!

**Review by john-rock — APPROVED** [review]
lgtm! Confirmed headers in `expandable` component are removed from ToC after loading.

**Pull request merged**